### PR TITLE
Changed progress indicator to use drawArc.

### LINKED ISSUE #3599: Circular progress indicators animate slowly
The material gallery [progress indicators demo] only manages about 30 fps, even with the non-standard elliptical progress indicator commented out.

The problem appears to be SKIA arc drawing - https://bugs.chromium.org/p/skia/issues/detail?id=5227

### PR DESCRIPTION
**Pull request by dragostis**
Using paths' arcTo was causing performance issues in the Gallery.
This commit uses Skia's optimized drawArc in order to improve
performance. Fixes #3599.

Tracking Skia issue:
  https://bugs.chromium.org/p/skia/issues/detail?id=5227

### CODE REVIEW

**Comment by dragostis** [comment]
Will fail until https://github.com/flutter/engine/pull/2995 is merged and the engine rolled.

File: packages/flutter/lib/src/material/progress_indicator.dart [hunk 1]
```diff
@@ -243,9 +243,10 @@ class _CircularProgressIndicatorPainter extends CustomPainter {
     if (value == null) // Indeterminate
       paint.strokeCap = StrokeCap.square;
 
-    Path path = new Path()
-      ..arcTo(Point.origin & size, arcStart, arcSweep, false);
-    canvas.drawPath(path, paint);
+    // Path path = new Path()
+    //   ..arcTo(Point.origin & size, arcStart, arcSweep, false);
+    // canvas.drawPath(path, paint);
```
Comment (Hixie) [inline packages/flutter/lib/src/material/progress_indicator.dart]: please do not check in commented-out code

Comment (dragostis) [inline packages/flutter/lib/src/material/progress_indicator.dart]: @Hixie Definitely not! This PR is just groundwork until I roll the engine. (which should happen soon)

**Comment by chinmaygarde** [comment]
lgrm

**Pull request merged**